src/api/types.ts

```diff
@@ -0,0 +1,85 @@
+export interface Exercise {
+    [key: string]: string | string[] | undefined | '' | 'kg' | 'lb',
```

Comment:
Why? I thought it can be only `kg | lb`, can it also be undefined?

Reply:
Hmm... it's hard to explain but the thing is I had troubles with types for this function (CreateView):
```js
const addExerciseValues = (e: { target: HTMLInputElement }) => {
  const key = e.target.name
  const value = e.target.value
  if (exerciseData.value[key]) {
    exerciseData.value[key] = value
  }
}
```
I had the following TS error without adding this first line: `[key: string]: string | string[] | undefined | '' | 'kg' | 'lb'` to the interface (solution provided by someone from stackoverflow):
![Screenshot_1](https://user-images.githubusercontent.com/113309294/228377564-dbf4121e-e710-4f41-a133-da8328658351.png)
I don't even know exactly what it does but apparently is types each key in this interface as a string and those all types you're asking for are actually all possible types in this interface.

Reply:
Because you should check if key exists in exerciseData I guess, the message is below 🙃

Reply:
Ehm... I believe I checked it in this `if` statement?

The message was there but it told me nothing.

However, I gave it some thought and realized maybe the problem lied in the object itself. I mean, in this `Exercise` object there are two keys that are supposed to have a value other than a string. So an input value, being a string cannot be assigned to those. So I believe I found a way around but I don't know if this is correct. The only thing I know here is that TS stopped yelling at me 😛

So, please check if it's correct now - if not, I'm afraid I have no idea how to make it work...